src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs

```diff
@@ -51,35 +51,67 @@ protected static bool CanUseSpeculativeSemanticModel(Document document, int posi
             return !service.GetMemberBodySpanForSpeculativeBinding(node).IsEmpty;
         }
 
+        internal static CompletionList GetCompletionList(CompletionListProvider provider, Document document, int position, CompletionTriggerInfo triggerInfo)
+        {
+            var itemsBuilder = ImmutableArray.CreateBuilder<CompletionItem>();
+            Action<CompletionItem> addCompletionItem = item =>
+            {
+                itemsBuilder.Add(item);
+            };
+
+            CompletionItem builder = null;
+            Action<CompletionItem> registerBuilder = item =>
+            {
+                builder = item;
+            };
+
+            var isExclusive = false;
+            Action<bool> makeExclusive = value =>
+            {
+                isExclusive = value;
+            };
+
+            var context = new CompletionListContext(document, position, triggerInfo, addCompletionItem, registerBuilder, makeExclusive, CancellationToken.None);
```

Comment:
Are there any contexts where we don't have the above behavior?  Should we just pull the above into CompletionListContext?